qualtran/cirq_interop/t_complexity_protocol.py

```diff
@@ -170,65 +178,55 @@ def _get_hash(val: Any, fail_quietly: bool = False):
 
 
 def _t_complexity_from_strategies(
-    stc: Any, fail_quietly: bool, strategies: Iterable[Callable[[Any, bool], Optional[TComplexity]]]
+    stc: Any, strategies: Iterable[Callable[[Any], Optional[TComplexity]]]
 ):
     ret = None
     for strategy in strategies:
-        ret = strategy(stc, fail_quietly)
+        ret = strategy(stc)
         if ret is not None:
             break
     return ret
 
 
 @cachetools.cached(cachetools.LRUCache(128), key=_get_hash, info=True)
 def _t_complexity_for_gate_or_op(
-    gate_or_op: Union[cirq.Gate, cirq.Operation, Bloq], fail_quietly: bool
+    gate_or_op: Union[cirq.Gate, cirq.Operation, Bloq]
 ) -> Optional[TComplexity]:
```

Comment:
This currently fails when `gate_or_op` is a `cirq.GateOperation` which wraps a `GateWithRegisters` that doesn't have an explicit annotation. This is because we currently fetch the underlying `gate` from an `op` via an `isinstance(stc, cirq.Operation)` only in `_from_explicit_annotation` 

I suggest we should add a top level check in `_t_complexity_for_gate_or_op` and run all the strategies on the underlying gate when `gate_or_op` is an `op`. 

```
    if isinstance(gate_or_op, cirq.Operation) and gate_or_op.gate is not None:
        return _t_complexity_from_gate(gate_or_op.gate)
```

Reply:
thanks, done